### PR TITLE
Remove migrated content from Whitehall search index

### PR DESCRIPTION
**Pull request by gclssvglx**
## User story

**As a** publisher
**I want** my content to be removed from Whitehall search index when it is migrated
**So that** the content doesn't appear in search twice

## What
We are adding a step to the migrated endpoint to remove the content for this document from the government index.

## Why

When content has been migrated from Whitehall to Content Publisher we need it to no longer exist in the Search API index for Whitehall content.

[Trello](https://trello.com/c/nJARKkob/1231-remove-content-from-whitehall-search-index)

### CODE REVIEW

File: app/controllers/admin/export/document_controller.rb [hunk 1]
```diff
@@ -33,6 +33,7 @@ def migrated
 
     document.editions.each do |edition|
       Whitehall::InternalLinkUpdater.new(edition).call
+      Whitehall::SearchIndex.delete(edition)
```
Comment (brucebolt) [inline app/controllers/admin/export/document_controller.rb]: It's likely only the most recent edition of the document will be present in the search index.  Since this is looping through all editions, do you know how `Whitehall::SearchIndex.delete` will behave for an edition that is not present in the search index?

File: .gitignore [hunk 2]
```diff
@@ -24,3 +24,4 @@
 /doc/
 /failures
 .DS_Store
+.byebug_history
```
Comment (brucebolt) [inline .gitignore]: Could this be added into a separate commit to ensure the commit message ties up with the code change?

Comment (gclssvglx) [reply .gitignore]: Created a separate commit for this.

File: app/controllers/admin/export/document_controller.rb (hunk 1 above)
Comment (gclssvglx) [reply app/controllers/admin/export/document_controller.rb]: Hi Bruce,

We've tested this (via console rails c) and it appears to behave exactly the same if the edition exists or not.  The method call delegates out to a worker job (SearchIndexDeleteWorker) that is queued.

When run the worker job calls `Whitehall::SearchIndex.for(index.to_sym, logger: logger).delete(link)` which will do nothing (return nil) if the edition is not in the index.

Think we're safe!
